### PR TITLE
feat: add mutual friends

### PR DESCRIPTION
**Pull request by Julieta11**
Retrieving the addresses of the friends that are shared between the authenticated user and the user specified in the request.

### CODE REVIEW

File: src/FriendsManagementClient.ts [hunk 1]
```diff
@@ -331,18 +345,36 @@ enum FriendshipEvent {
...
             return response.friendships.map(f => f.address)
-        } catch(e) {
+        } catch (e) {
             console.error(e)
         }
     }
     return []
 }
 
+export async function getFriendsFromSocialService(baseUrl: string, userId: string, auth: string): Promise<string[]> {
```
Comment (moliva) [inline src/FriendsManagementClient.ts:362]: Let's use this opportunity to add docs to this function as well.

File: test/unit/socialServerMock.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    // No friends userId
+    server.get(`/v1/friendships/${userId}`, (req, res) => {
+        if (isValidToken(req)) {
+            res.jsonp({
+                friendships: []
+            })
+        } else {
+            res.sendStatus(401)
+        }
```
Comment (moliva) [inline test/unit/socialServerMock.ts:14]: Let's wrap the logic of the `if (isValidToken) { ... } else { res.sendStatus(401) }` as another function. WDYT?

File: src/FriendsManagementClient.ts [hunk 3]
```diff
@@ -331,18 +345,60 @@ enum FriendshipEvent {
...
 }
 
+/**
+ * Get friends addresses list of the logged-in user.
+ * @param baseUrl - service base url.
+ * @param userId - user id of the logged-in user.
+ * @param auth - access token associated with this account.
+ */
+export async function getFriendsFromSocialService(baseUrl: string, userId: string, auth: string): Promise<string[]> {
+    const url = new URL(`${baseUrl}/v1/friendships/${userId}`)
```
Comment (agusaldasoro) [inline src/FriendsManagementClient.ts:387]: can you change this to use `/friendships/me`? So we don't have to send the userId as param

**Review by agusaldasoro — CHANGES_REQUESTED** [review]

**Review by jmoguilevsky — APPROVED** [review]

**Pull request merged**